src/posts/2022-07-13-L4LB-XDP/index.md

```diff
@@ -0,0 +1,225 @@
+---
+path: '/blog/2022/04/12/cilium-standalone-L4LB-XDP'
+date: '2022-07-13T17:00:00.000Z'
+title: 'Cilium Standalone Layer 4 Load Balancer XDP'
+isPopular: true
+isFeatured: false
+ogImage: cpu+.png
+ogSummary: 'See the performance increase Seznam.cz saw using Cilium for load balancing'
+categories:
+  - Community
+tags:
+  - Cilium
+  - Clustermesh
+---
+
+![Seznam.cz logo](Seznam.cz_logo.png)
+
+_July 13th, 2022_  
+_Author: Ondrej Blazek, Infrastructure Engineer @ Seznam.cz_
+
+Seznam.cz is a Czech technological company developing a custom search engine, advertising platform, online maps, content management system, and also private cloud services, custom hardware, and datacenters.
+ 
+ 
+# Architecture
+ 
+Seznam's infrastructure historically used F5 hardware load balancers but we switched to software load balancers a few years ago. Up until now we've been using a [multiple tier](https://vincent.bernat.ch/en/blog/2018-multi-tier-loadbalancer) setup - [ECMP routing](https://vincent.bernat.ch/en/blog/2018-multi-tier-loadbalancer#first-tier-ecmp-routing) as the first tier + [IPVS](http://www.linuxvirtualserver.org/software/ipvs.html) as the second tier [L4 load balancer (L4LB)](https://vincent.bernat.ch/en/blog/2018-multi-tier-loadbalancer#second-tier-l4-load-balancing) + [Envoy proxy](https://www.envoyproxy.io/) as the third tier [L7 load balancer](https://vincent.bernat.ch/en/blog/2018-multi-tier-loadbalancer#last-tier-l7-load-balancing).  Unfortunately as traffic increased and, thanks to COVID, we started running short on hardware supplies, we had to look for alternatives to use our hardware more effectively.
+ 
+We have been watching Cilium closely and noticed the [announcement of maglev in Cilium](https://cilium.io/blog/2020/11/10/cilium-19#maglev) together with [`Standalone L4LB XDP`](https://cilium.io/blog/2021/05/20/cilium-110#standalonelb) announced in 1.10. XDP hook is known for its efficient use of CPU and has an extremely high performance. This was very interesting for our team because we have been hitting traffic spikes up to 20M active connections which was massively increasing CPU usage of our IPVS nodes.
+ 
+ 
+Our load balancer setup directs external traffic into the Kubernetes and OpenStack clusters and IPVS is used in a classic "load balancer on a stick" scenario. In a simplified world this looks like this:
+ 
+![Seznam.cz architecture](architecture.svg)
+ 
+Since, we have been using the maglev scheduler (which is part of a netfilter inside the Linux kernel since v4.18) the `Standalone L4LB XDP` was a perfect match to try as an alternative because it supports all the main features we required (IPIP, DSR, maglev).
```

Comment:
```suggestion
Since, we have been using the maglev scheduler (which is part of a netfilter inside the Linux kernel since v4.18) the **Standalone L4LB XDP** was a perfect match to try as an alternative because it supports all the main features we required (IPIP, DSR, maglev).
```